### ISSUE TITLE
Type argument inference fails for a generic call on a new array inside a lambda passed to a generic method

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2086]
Type argument inference fails on this code. javac alone compiles it.

```java
import java.util.function.Supplier;

class V6 {
    static <R> R run(Supplier<R> s) { throw new Error(); }
    static <T> T[] arr(T[] a) { return a; }

    String[] m() { return run(() -> arr(new String[0])); }
}
```

```
javac -processor regex -AconvertTypeArgInferenceCrashToWarning=false V6.java
```

```
error:  False bound for: Constraint: inference type: T[] <: R extends Object Result: FALSE
```

With the default options nothing is printed. The test harness always passes `-AconvertTypeArgInferenceCrashToWarning=false`, so checker tests see this failure. The same error occurs with `-processor nullness`.

| Variant | Result |
|---|---|
| as above | fails |
| `run(() -> lst(new String[0]))`, where `lst` returns `List<T>` | fails |
| `run2(arr(new String[0]))`, no lambda | ok |
| `run(() -> arr(a))`, with a local array `a` | ok |
| `run(() -> id(new String[0]))`, where `id` returns `T` | ok |

Expected: no error.

Environment: master `3d21be4eec`, JDK 25.
